File: apps/backend/app/schemas/pull_request.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class PullRequestRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    workflow_id: UUID
    repo_owner: str | None = None
    repo_name: str | None = None
    branch_name: str | None = None
    pr_number: int | None = None
    pr_url: str | None = None
    status: str
    title: str
    description: str | None = None
    changed_files: list[dict[str, Any]] | None = None
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

File: apps/frontend/src/types/pull-request.ts
export interface PullRequestHistory {
  id: string;
  workflow_id: string;
  repo_owner?: string | null;
  repo_name?: string | null;
  branch_name?: string | null;
  pr_number?: number | null;
  pr_url?: string | null;
  status: string;
  title: string;
  description?: string | null;
  changed_files?: Array<Record<string, unknown>> | null;
  error_message?: string | null;
  created_at: string;
  updated_at: string;
}

export interface PRSafetyCheck {
  can_open_pr: boolean;
  workflow_id: string;
  pr_preview_id?: string | null;
  existing_pr_url?: string | null;
  prepared_files: Array<{
    path: string;
    original_path: string;
    artifact_type: string;
  }>;
  checks: Array<{
    name: string;
    passed: boolean;
    message: string;
  }>;
}
